CONTRIBUTING.rst

```diff
@@ -461,52 +494,159 @@ Step 5: Pass PR Review
 these guidelines:
 
--   Include tests, either as doctests, unit tests, or both, to your pull
-    request.
+-   Include tests, either as doctests, unit tests, or both, to your pull request.
 
     The airflow repo uses `GitHub Actions <https://help.github.com/en/actions>`__ to
     run the tests and `codecov <https://codecov.io/gh/apache/airflow>`__ to track
     coverage. You can set up both for free on your fork. It will help you make sure you do not
     break the build with your PR and that you help increase coverage.
+    Also we advise to install locally `pre-commit hooks <STATIC_CODE_CHECKS.rst#pre-commit-hooks>`__ to
+    apply various checks, code generation and formatting at the time you make a local commit - which
+    gives you near-immediate feedback on things you need to fix before you push your code to the PR, or in
+    many case it will even fix it for you locally so that you can add and commit it straight away.
 
--   Follow our project's `Coding style and best practices`_.
+-   Follow our project's `Coding style and best practices`_. Usually we attempt to enforce the practices by
+    having appropriate pre-commits. There are checks amongst them that aren't currently enforced
+    programmatically (either because they are too hard or just not yet done).
 
-    These are things that aren't currently enforced programmatically (either because they are too hard or just
-    not yet done.)
+-   We prefer that you ``rebase`` your PR (and do it quite often) rather than merge. It leads to
+    easier reviews and cleaner changes where you know exactly what changes you've done. You can learn more
+    about rebase vs. merge workflow in `Rebase and merge your pull request <https://github.blog/2016-09-26-rebase-and-merge-pull-requests/>`__
+    and `Rebase your fork <http://stackoverflow.com/a/7244456/1110993>`__. Make sure to resolve all conflicts
+    during rebase.
 
--   `Rebase your fork <http://stackoverflow.com/a/7244456/1110993>`__, and resolve all conflicts.
+-   When merging PRs, Maintainer will use **Squash and Merge** which means then your PR will be merged as one
+    commit, regardless of the number of commits in your PR. During the review cycle, you can keep a commit
+    history for easier review, but if you need to, you can also squash all commits to reduce the
+    maintenance burden during rebase.
 
--   When merging PRs, Committer will use **Squash and Merge** which means then your PR will be merged as one commit, regardless of the number of commits in your PR. During the review cycle, you can keep a commit history for easier review, but if you need to, you can also squash all commits to reduce the maintenance burden during rebase.
+-   Add an `Apache License <http://www.apache.org/legal/src-headers.html>`__ header to all new files. If you
+    have ``pre-commit`` installed, pre-commit will do it automatically for you. This is one of the reasons
+    why we recommend using ``pre-commit``.
```

Comment:
I'd leave out the third sentence to make it shorter - but don't take this comment too serious :-D Just assuming somebody has read the text above.
```suggestion
-   Add an `Apache License <http://www.apache.org/legal/src-headers.html>`__ header to all new files. If you
    have ``pre-commit`` installed, pre-commit will do it automatically for you.
```

Reply:
I reworded it. My main point is that `pre-commit` saves you from all those kind of mundane and repetitive things that people are wining about. For me `pre-commit` adding licence headers is THE huge win that justifies the little slow-down when committing.